OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -101,7 +101,10 @@ public FrozenActor(Actor actor, PPos[] footprint, Player viewer, bool startsReve
 		public void RefreshState()
 		{
 			Owner = actor.Owner;
-			TargetTypes = actor.GetEnabledTargetTypes().ToHashSet();
+
+			// PERF: Reuse collection to avoid allocations.
+			TargetTypes.Clear();
+			TargetTypes.UnionWith(actor.GetEnabledTargetTypes());
```

Comment:
@RoosterDragon TargetTypes seems like a great candidate for load-time bitsetification.

Reply:
Interesting class. Is the idea you specify `Bits<SomeMarkerStruct> TargetTypes` in multiple places and it automatically converts everything as it's loaded?